threadpoolctl.py

```diff
@@ -599,6 +211,529 @@ def get_original_num_threads(self):
         if warning_apis:
             warnings.warn(
                 "Multiple value possible for following user apis: "
-                + ', '.join(warning_apis) + ". Returning the minimum.")
+                + ", ".join(warning_apis) + ". Returning the minimum.")
 
         return num_threads
+
+    def _check_params(self, limits, user_api):
+        """Suitable values for the _limits, _user_api and _prefixes attributes
+        """
+        if limits is None or isinstance(limits, int):
+            if user_api is None:
+                user_api = _ALL_USER_APIS
+            elif user_api in _ALL_USER_APIS:
+                user_api = [user_api]
+            else:
+                raise ValueError(
+                    "user_api must be either in {} or None. Got "
+                    "{} instead.".format(_ALL_USER_APIS, user_api))
+
+            if limits is not None:
+                limits = {api: limits for api in user_api}
+            prefixes = []
+        else:
+            if isinstance(limits, list):
+                # This should be a list of dicts of modules, for compatibility
+                # with the result from threadpool_info.
+                limits = {module["prefix"]: module["num_threads"]
+                          for module in limits}
+            elif isinstance(limits, _ThreadpoolInfo):
+                # To set the limits from the modules of a _ThreadpoolInfo
+                # object.
+                limits = {module.prefix: module.num_threads
+                          for module in limits}
+
+            if not isinstance(limits, dict):
+                raise TypeError("limits must either be an int, a list or a "
+                                "dict. Got {} instead".format(type(limits)))
+
+            # With a dictionary, can set both specific limit for given modules
+            # and global limit for user_api. Fetch each separately.
+            prefixes = [prefix for prefix in limits if prefix in _ALL_PREFIXES]
+            user_api = [api for api in limits if api in _ALL_USER_APIS]
+
+        return limits, user_api, prefixes
+
+    def _set_threadpool_limits(self):
+        """Change the maximal number of threads in selected thread pools.
+
+        Return a list with all the supported modules that have been found
+        matching `self._prefixes` and `self._user_api`.
+        """
+        if self._limits is None:
+            return None
+        else:
```

Comment:
nit: I usually prefer to not have the `else` in this case as the body of the function is included in it.